### PR TITLE
SDI-71: 📝 Add spring docs generation and test

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public Agency updateAgency(@PathVariable("agencyId") @Parameter(description = "T
     }
 
     @ApiResponses({
-        @ApiResponse(responseCode = "201", description = "The Agency location created", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = Agency.class))}),
+        @ApiResponse(responseCode = "201", description = "The Agency location created"),
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/AgencyResource.java:150]: content not required as can be deduced from return type

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 2]
```diff
@@ -948,12 +954,12 @@ public AdjudicationSummary getAdjudicationSummary(@PathVariable("bookingId") @Pa
...
         @RequestParam(value = "visitStatus", required = false) @Parameter(name = "Status of visit. code from VIS_COMPLETE domain, e.g: CANC (Cancelled) or SCH (Scheduled)", example = "SCH") final String visitStatus,
         @RequestParam(value = "cancellationReason", required = false) @Parameter(name = "Reason for cancellation. code from MOVE_CANC_RS domain, e.g: VISCANC (Visitor Cancelled) or NO_VO (No Visiting Order)", example = "NSHOW") final String cancellationReason,
         @RequestParam(value = "prisonId", required = false) @Parameter(description = "The prison id", example = "MDI") final String prisonId,
-        @RequestParam(value = "page", required = false) @Parameter(description = "Target page number, zero being the first page") final Integer pageIndex,
-        @RequestParam(value = "size", required = false) @Parameter(description = "The number of results per page") final Integer pageSize) {
+        @RequestParam(value = "page", required = false, defaultValue = "0") @Parameter(description = "Target page number, zero being the first page") final Integer pageIndex,
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:961]: setting default values here means swagger parameter has those values as default too

File: src/test/resources/application-test.yml [hunk 3]
```diff
@@ -1,3 +1,7 @@
 management.endpoint:
   health.cache.time-to-live: 0
   info.cache.time-to-live: 0
+
+spring:
+  codec:
+    max-in-memory-size: 10MB
```
Comment (petergphillips) [inline src/test/resources/application-test.yml:7]: to cope with large v3/api-docs response

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**